src/legacy/server/index_patterns/mixin.ts

```diff
@@ -17,25 +17,38 @@
  * under the License.
  */
 
+import Hapi from 'hapi';
+import { APICaller } from 'src/core/server';
+import { Legacy } from 'kibana';
+import KbnServer from 'src/legacy/server/kbn_server';
 import { IndexPatternsService } from './service';
 
-import {
-  createFieldsForWildcardRoute,
-  createFieldsForTimePatternRoute,
-} from './routes';
+import { createFieldsForWildcardRoute, createFieldsForTimePatternRoute } from './routes';
 
-export function indexPatternsMixin(kbnServer, server) {
-  const pre = {
+export type IndexPatternsServiceFactory = (args: {
+  callCluster: (endpoint: string, clientParams: any, options: any) => Promise<any>;
+}) => IndexPatternsService;
+
+interface IndexPatternRequest extends Legacy.Request {
+  getIndexPatternsService: () => IndexPatternsService;
+}
+
+interface ServerWithFactory extends Legacy.Server {
+  addMemoizedFactoryToRequest: (...args: any) => void;
+}
+
+export function indexPatternsMixin(kbnServer: KbnServer, server: ServerWithFactory) {
+  const pre: Record<string, Hapi.RouteOptionsPreAllOptions> = {
     /**
-    *  Create an instance of the `indexPatterns` service
-    *  @type {Hapi.Pre}
-    */
+     *  Create an instance of the `indexPatterns` service
+     *  @type {Hapi.Pre}
+     */
     getIndexPatternsService: {
       assign: 'indexPatterns',
-      method(request) {
+      method(request: IndexPatternRequest) {
         return request.getIndexPatternsService();
-      }
-    }
+      },
+    },
   };
```

Comment:
@lukeelmers There's an error I can't figure about the Request types here, can you take a look?